src/common/errors_js.ml

```diff
@@ -106,6 +201,21 @@ let file_of_error err =
   let loc = loc_of_error err in
   match loc.Loc.source with Some filename -> filename | None -> ""
 
+(* TODO: Is this actually needed? Does it make sense with the new format? *)
+let remove_newlines (color, text) =
+  (color, Str.global_replace (Str.regexp "\n") "\\n" text)
+
+let print_error_color ~one_line ~color error =
+  let level, messages, trace_reasons = error in
+  let messages = append_trace_reasons messages trace_reasons in
+  let messages = merge_comments_into_blames messages in
+  let header = print_error_header (List.hd messages) in
+  let main_file = file_of_error error in
+  let formatted_messages = List.map (print_message_nice main_file) messages in
```

Comment:
I think you're relying on a wrong assumption here: you're assuming that all messages point to the same file, but that's not necessarily true. You could easily have an error span multiple files. In which case the format needs to be:
`file1`
`code line` <- message
`file2`
`code line` <- message

etc.

You could still omit printing out a file name if it matches the previous one, which will make the output look like what you have now when you're dealing with errors within the same file.

Anyway, what this means eventually is that you can't always pass `main_file` to `print_message_nice`, you need to extract the file out from every message.